content/docs/esc/other-integrations/cloudflare.md

```diff
@@ -0,0 +1,254 @@
+
+- Add an ESC Environment to your Pulumi stack
+- Assign an ESC Secret to Cloudflare resources
+- Consume the ESC Secret in a Worker script
+
+### Additional Prerequisites
+
+In addition to the [prerequisites above](#prerequisites), ensure you have:
+
+- installed [Pulumi CLI](https://www.pulumi.com/docs/install/)
+- installed a [Pulumi-supported programming language](https://www.pulumi.com/docs/languages-sdks/)
+
+{{< notes type="info" >}}
+**New to Pulumi IaC?** Complete the [Getting Started tutorial](https://developers.cloudflare.com/pulumi/tutorial/hello-world/).
+{{< /notes >}}
+
+### 1. Create (or Modify) an ESC Environment
+
+Use the Pulumi ESC CLI to create and configure an Environment. Alternatively, to use the Pulumi Cloud console follow the [console instructions](https://www.pulumi.com/docs/esc/get-started/create-environment/#create-via-the-console).
+
+```bash
+esc login # if needed
+
+## create a new ESC Environment
+ESC_ENV=my-pulumi-environment
+esc env init ${ESC_ENV}
+```
+
+Paste the contents below in the editor and replace the values. These values are to be consumed by a Pulumi program hence they are placed under the `pulumiConfig` section. See the [syntax reference](https://www.pulumi.com/docs/esc/reference/) for more options.
+
+```yaml
+values:
+  pulumiConfig:
+    CLOUDFLARE_API_TOKEN:
+      fn::secret: abc123abc123abc123abc123abc123
+    CLOUDFLARE_ACCOUNT_ID: abc123abc123
+    TOP_SECRET:
+      fn::secret: "aliens are real"
+```
+
+### 2. Add ESC to a Pulumi Stack
+
+You'll create a new stack to test the changes in isolation. Optionally, use an existing Stack.
+
+Inside your Pulumi project directory, run:
+
+```bash
+pulumi stack init my-esc-stack
+pulumi config env add ${ESC_ENV}
+```
+
+### 3. Assign an ESC Secret to a [Worker Secret Binding](https://www.pulumi.com/registry/packages/cloudflare/api-docs/workerscript/#inputs)
+
+With the ESC Environment referenced in the Stack, you'll be able to consume ESC values to assign them to a Secret Binding Input. Here is an example Pulumi program written in TypeScript:
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as cloudflare from "@pulumi/cloudflare";
+import * as fs from "fs";
+
+const config = new pulumi.Config();
+
+// Create a Cloudflare WorkerScript with a secret binding
+const workerScript = new cloudflare.WorkerScript("myWorkerScript", {
+    name: "my-worker-script",
+    content: fs.readFileSync("worker.ts", "utf8"),
+    secretTextBindings: [
+        {
+            name: "TOP_SECRET",
+            text: config.requireSecret("TOP_SECRET"),
+        },
+    ],
+});
+```
+
+The secret binding configuration allows the Worker to consume the secret in the `fetch` handler. Here's the contents of the `worker.ts` file.
+
+```typescript
+export interface Env {
+  TOP_SECRET: string;
+}
+export default {
+  async fetch(request, env, ctx): Promise<Response> {
+    return new Response(`Did you know that "${env.TOP_SECRET}"?`);
+  }
+}
+```
```

Comment:
I'd check with Cleve, but this scenario feels like SDK is an optimal solution than embedding as part of pulumiConfig

Reply:
@cleverguy25 thoughts on using the SDK here instead?